app/models/labware_creators/tagged_plate.rb

```diff
@@ -20,6 +20,20 @@ class TaggedPlate < Base
     self.attributes += [:tag_plate_barcode, { tag_plate: %i[asset_uuid template_uuid] }]
     self.default_transfer_template_name = 'Custom pooling'
 
+    # The TaggedPlate labware creator uses API V1 Plate as parent. The V1 Plate
+    # has the populate_wells_with_pool method that assigns pools to wells using
+    # its pools attribute. However, this is incompatible with the V2 API Plate.
+    # The V2 plate does not have such a method and its pool attribute has a
+    # different structure. The populate_wells_with_pool method is called in the
+    # initialize method of the TaggedPlate labware creator on the parent plate.
+    # This makes it difficult to subclass the labware creator and also use a V2
+    # API plate as parent. If the class attribute should_populate_wells_with_pool
+    # is set to false, it will not call the populate_wells_with_pool method on
+    # the parent plate.
+
+    class_attribute :should_populate_wells_with_pool
+    self.should_populate_wells_with_pool = true # parent is a V1 plate
+
```

Comment:
I think I've been dealing with this.  See https://github.com/sanger/limber/blob/Y24-190-use-api-v2-plate-everywhere/app/sequencescape/sequencescape/api/v2/plate.rb#L136-L144

Reply:
@sdjmchattie I have done that before at https://github.com/sanger/limber/pull/1991/commits/daf8849548cedce6c8b76cd61925a7f1b45a4049 . I think the code looks slightly different but the idea is the same.

However, I ended up with the situation that while V1 plate has no pools, V2 plate has pools for the same underlying plate from Sequencescape, which is not correct. I think some refactoring and migration were done and pools are handled differently in V2 and they have different layout. I have spent too much time on it, I had to move on as I can't find why a specific pool is assigned on a well. I think when you have multiple submissions; the assignment in the workaround has an unpredictable outcome in that loop by finding the first match.

Specific to my PR, I don't need the pool on the well and I need to subclass TaggedPlate labware creator, so I am making the subclass to jump over that line because I need the parent initialiser called otherwise.

As a side note, I have found submission_pools association more relevant and compatible. I have added that resource in an SS  PR (merged now).

Reply:
API changes are handled separately as part of v1 to v2 migration. I will merge the PR and look back if necessary. Resolving conversation.